internal/summary/summary.go

```diff
@@ -24,29 +33,87 @@ func NewSummary() *Summary {
 }
 
 // Gather collates data into Summary
-func (s *Summary) Gather(services []model.Service) {
+func (s *Summary) Gather(services []model.Service) error {
 
 	for _, svc := range services {
-		var path string
 		// Assume that, in practice, there is only one mount
+		path := ""
 		if len(svc.Mounts) > 0 {
 			path = svc.Mounts[0].Repo.LocalPath
-		} else {
-			path = ""
 		}
-		s.Services = append(s.Services, &Service{
-			Name: string(svc.Name),
-			Path: path,
-		})
+
+		svcSummary := &Service{
+			Name:    string(svc.Name),
+			Path:    path,
+			k8sData: []k8sDatum{},
+		}
+
+		kubeDatum := &k8sDatum{}
+
+		entities, err := k8s.ParseYAMLFromString(svc.K8sYaml)
+		if err != nil {
+			return err
+		}
+
+		lbs := k8s.ToLoadBalancers(entities)
+
+		for _, lb := range lbs {
+			kubeDatum.LoadBalancer = append(kubeDatum.LoadBalancer, lb)
+		}
+
+		for _, e := range entities {
+			kubeDatum.Group = e.Kind.Group
+			kubeDatum.Version = e.Kind.Version
+			kubeDatum.Kinds = append(kubeDatum.Kinds, e.Kind.Kind)
+		}
+
+		svcSummary.k8sData = append(svcSummary.k8sData, *kubeDatum)
```

Comment:
...wait, so a given service summary... will only ever have a single datum attached?

Reply:
Clearly all of this was an elaborate scheme to demonstrate "data vs datum" pedantry. I'm glad you flagged it because in the end I really don't need `[]k8sDatum` at all. Not sure why I thought I did.